postgres-appliance/scripts/test_reload_ssl.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+#
+# This script is being called every TEST_INTERVAL_MINUTES to check if the
+# spilo or postgres configuration need to be reloaded.
+#
+# Usage: test_reload_ssl.sh <TEST_INTERVAL_MINUTES>
+set -euo pipefail
+
+# How often this script is being called
+test_interval_min=$1
+test_interval_sec=$((test_interval_min * 60))
+
+## Functions ##
+
+debug() {
+  echo "$*" >&2
+}
+
+has_changed_in_interval() {
+    local file=$1
+    if [[ -z "$file" ]]; then
+        debug "argument is empty"
+        return 2
+    fi
+    if [[ ! -e "$file" ]]; then
+        debug "$file does not exist"
+        return 2
+    fi
+    local mtime now elapsed_sec
+    mtime=$(stat -c '%Y' "$file")
+    now=$(date +%s)
+    elapsed_sec=$(( now - 1 - mtime ))
+    [[ $elapsed_sec -lt $test_interval_sec ]]
+}
+
+## Main ##
+
+if has_changed_in_interval "$SSL_PRIVATE_KEY_FILE"; then
```

Comment:
You [setup](https://github.com/zalando/spilo/pull/373/files#diff-5e29afdf8025daafbc08cc254eb83f9fR787) a cron job based on `SSL_CERTIFICATE_FILE`, but here you check the `SSL_PRIVATE_KEY_FILE`, which is rarely supposed to be rotated.

Reply:
doh. It now checks for an independent `SSL_TEST_RELOAD` attribute, which is set to True if `SSL_CERTIFICATE_FILE` is set or False otherwise to allow more flexibility and expose the logic.